pkg/k8scontext/k8scontext_test.go

```diff
@@ -28,51 +72,119 @@ var _ = Describe("K8scontext", func() {
 		// Create the mock K8s client.
 		k8sClient = testclient.NewSimpleClientset()
 
-		// Create the "test-ingress-controller" namespace.
-		// We will create all our resources under this namespace.
-		ns := &v1.Namespace{
-			ObjectMeta: metav1.ObjectMeta{
-				Name: ingressNS,
-			},
-		}
 		_, err := k8sClient.CoreV1().Namespaces().Create(ns)
 
 		Expect(err).Should(BeNil(), "Unable to create the mock Kuberentes client: %v", err)
 
+		_, err = k8sClient.Extensions().Ingresses(ingressNS).Create(ingress)
+		Expect(err).Should(BeNil(), "Unabled to create ingress resource due to: %v", err)
+
 		// Create a `k8scontext` to start listiening to ingress resources.
-		ctxt = k8scontext.NewContext(k8sClient, ingressNS, 30*time.Second)
+		ctxt = k8scontext.NewContext(k8sClient, ingressNS, 1000*time.Second)
 		Expect(ctxt).ShouldNot(BeNil(), "Unable to create `k8scontext`")
 	})
 
 	Context("Checking if we are able to listen to Ingress Resources", func() {
 		It("Should be able to retrieve all Ingress Resources", func() {
-			// Create the Ingress resource.
-			ingress := &v1beta1.Ingress{
-				Spec: v1beta1.IngressSpec{
-					Backend: &v1beta1.IngressBackend{
-						ServiceName: "hello-world",
-						ServicePort: intstr.IntOrString{
-							Type:   intstr.Int,
-							IntVal: 80,
-						},
-					},
-				},
-			}
-
-			// l.Println("Running the ingress test !!")
-			_, err := k8sClient.Extensions().Ingresses(ingressNS).Create(ingress)
-			Expect(err).Should(BeNil(), "Unabled to create ingress resource due to: %v", err)
-
 			// Retrieve the Ingress to make sure it was created.
 			ingresses, err := k8sClient.Extensions().Ingresses(ingressNS).List(metav1.ListOptions{})
 			Expect(err).Should(BeNil(), "Unabled to retrieve stored ingresses resource due to: %v", err)
 			Expect(len(ingresses.Items)).To(Equal(1), "Expected to have a single ingress stored in mock K8s but found: %d ingresses", len(ingresses.Items))
 
-			// Start the informers.
+			// Start the informers. This will sync the cache with the latest ingress.
+			ctxt.Run()
+
+			ingressListInterface := ctxt.Caches.Ingress.List()
+			Expect(len(ingressListInterface)).To(Equal(1), "Expected to have a single ingress in the cache but found: %d ingresses", len(ingressListInterface))
+
+			// Retrive the ingresses learnt by the controller.
+			testIngresses := ctxt.GetHTTPIngressList()
+			Expect(len(testIngresses)).To(Equal(1), "Expected to have a single ingress in the k8scontext but found: %d ingresses", len(testIngresses))
+
+			// Make sure the ingress we got is the ingress we stored.
+			Expect(testIngresses[0]).To(Equal(ingress), "Expected to retrieve the same ingress that we inserted, but it seems we found the following ingress: %v", testIngresses[0])
```

Comment:
question: does v1beta1.Ingress satisfy the GomegaMatcher  interface ? (making it possible to with Equal())

Reply:
Yeah. Its pretty intelligent. I haven't looked at the internals, but I think it uses the `reflect` package to do deep inspection to equate properties of the two objects. Similar to how `deepcopy` works using the reflect package.